AutoFinance.Broker/InteractiveBrokers/Controllers/TwsControllerBase.cs

```diff
@@ -849,55 +858,60 @@ public Task<List<PositionStatusEventArgs>> RequestPositions(CancellationToken ca
         /// Set the type for the market data feed
         /// </summary>
         /// <param name="marketDataTypeId">The feed level</param>
-        public void RequestMarketDataType(int marketDataTypeId)
+        /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
+        public Task RequestMarketDataTypeAsync(int marketDataTypeId)
         {
-            this.clientSocket.RequestMarketDataType(marketDataTypeId);
+            // Set the operation to cancel after 5 seconds
+            CancellationTokenSource tokenSource = new CancellationTokenSource(5000);
+            return this.RequestMarketDataTypeAsync(marketDataTypeId, tokenSource.Token);
         }
 
         /// <summary>
-        /// Get the PnL of the account.
+        /// Set the type for the market data feed
         /// </summary>
-        /// <param name="accountCode">The account code</param>
-        /// <param name="modelCode">The model code</param>
-        /// <returns>The PnL account update event from TWS.</returns>
-        public Task<PnLEventArgs> RequestPnL(
-            string accountCode,
-            string modelCode)
+        /// <param name="marketDataTypeId">The feed level</param>
+        /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
+        public Task RequestMarketDataTypeAsync(int marketDataTypeId, CancellationToken cancellationToken)
         {
-            // Set the operation to cancel after 5 seconds
-            CancellationTokenSource tokenSource = new CancellationTokenSource(5000);
-            return this.RequestPnL(accountCode, modelCode, tokenSource.Token);
+            string value = string.Empty;
+
+            var taskSource = new TaskCompletionSource<bool>();
+            this.twsCallbackHandler.MarketDataTypeEvent += (sender, args) =>
+            {
+                taskSource.TrySetResult(true);
+            };
+
+            cancellationToken.Register(() =>
+            {
+                taskSource.TrySetCanceled();
+            });
+
+            this.clientSocket.RequestMarketDataType(marketDataTypeId);
+
+            return taskSource.Task;
         }
 
         /// <summary>
         /// Get the PnL of the account.
         /// </summary>
         /// <param name="accountCode">The account code</param>
         /// <param name="modelCode">The model code</param>
-        /// <param name="cancellationToken">The cancellation token used to cancel the request</param>
         /// <returns>The PnL account update event from TWS.</returns>
         public Task<PnLEventArgs> RequestPnL(
             string accountCode,
-            string modelCode,
-            CancellationToken cancellationToken)
+            string modelCode)
```

Comment:
Make an integration test around demonstrating how to use it (requestPnl + cancelPnl, requestPnlSingle, cancelPnlSingle)

Reply:
Done.